mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp

```diff
@@ -383,13 +432,55 @@ class AffineParallelLowering : public OpRewritePattern<AffineParallelOp> {
     for (Attribute step : op.steps())
       steps.push_back(rewriter.create<ConstantIndexOp>(
           loc, step.cast<IntegerAttr>().getInt()));
-    // Creating empty scf.parallel op body with appropriate bounds.
-    auto parallelOp = rewriter.create<scf::ParallelOp>(loc, lowerBoundTuple,
-                                                       upperBoundTuple, steps);
-    rewriter.eraseBlock(parallelOp.getBody());
-    rewriter.inlineRegionBefore(op.region(), parallelOp.region(),
-                                parallelOp.region().end());
-    rewriter.eraseOp(op);
+    // Get the terminator operands.
+    Operation *affineParOpTerm = op.region().front().getTerminator();
+    SmallVector<Value, 4> terminatorOperands(
+        affineParOpTerm->getOperands().begin(),
+        affineParOpTerm->getOperands().end());
+    scf::ParallelOp par;
+    if (op.results().size() == 0) {
+      // Case with no reduction operations/return values.
+      par = rewriter.create<scf::ParallelOp>(loc, lowerBoundTuple,
+                                             upperBoundTuple, steps, nullptr);
+      rewriter.eraseBlock(par.getBody());
+      rewriter.inlineRegionBefore(op.region(), par.region(),
+                                  par.region().end());
+    } else {
+      // Case with multiple reduction operations/return values.
```

Comment:
A comment here would help saying that `scf.parallel` unfortunately uses a different modeling for reductions (unlike `scf.for` and `affine.for/affine.parallel`.